src/apiutils.jl

```diff
@@ -72,16 +72,36 @@ end
 
 function seed!(duals::AbstractArray{Dual{T,V,N}}, x,
                seed::Partials{N,V} = zero(Partials{N,V})) where {T,V,N}
-    for idx in structural_eachindex(duals, x)
-        duals[idx] = Dual{T,V,N}(x[idx], seed)
+    if isbitstype(V)
+        for idx in structural_eachindex(duals, x)
+            duals[idx] = Dual{T,V,N}(x[idx], seed)
+        end
+    else
+        for idx in structural_eachindex(duals, x)
+            if isassigned(x, idx)
+                duals[idx] = Dual{T,V,N}(x[idx], seed)
+            else
+                Base._unsetindex!(duals, idx)
```

Comment:
:/, this is quite unfortunate, is this really the only way to fix this?

Reply:
I think the idea is that if `x[idx]` is `#undef`, we want `duals[idx]` to be the same

Reply:
Yeah, I wasn't 100% percent happy but it was the best I could think of and what apparently is used in base. IMO we really want to ensure here that we get the same behaviour as `copyto!` on the primal part, to avoid that `yduals` contains any surprising values, possibly from previous chunks or differentations.

Reply:
```julia
julia> x = BigFloat[1, 2];

julia> y = similar(x); y[2] = 1;

julia> x
2-element Vector{BigFloat}:
 1.0
 2.0

julia> y
2-element Vector{BigFloat}:
 #undef
   1.0

julia> copyto!(x, y);

julia> x
2-element Vector{BigFloat}:
 #undef
   1.0
```